blog/2025/pcsx2-2.6/index.mdx

```diff
@@ -0,0 +1,606 @@
+
+<Image src={require("./img/benches/13305-Jak-pt2.webp").default} />
+
+<Image src={require("./img/benches/13305-SH-pt3.webp").default} />
+
+#### But wait there's more!
+
+A while back, we added extra barriers to fix some bugs in Time Crisis. In [#13595](https://github.com/PCSX2/pcsx2/pull/13595) TellowKrinkle discovered that those barriers were getting added on all GPU vendors. Upon further inspection, it appears as an NVIDIA driver bug, rather than actual issues with our OpenGL API usage. This allowed us to cut down the barriers on OpenGL on other GPU vendors by a lot.
+
+Here are the top 5 highest barrier reductions we have found:
+
+```ini
+WRC_4_SCES-52389_20230329024724
+Barriers: -6604 [13473=>6869]
+
+nfsmw_reflections
+Barriers: -2125 [4250=>2125]
+
+007 Agent Under Fire - door blend
+Barriers: -1646 [3284=>1638]
+
+Ace_Combat_04_-_Shattered_Skies_SLUS-20152_20230214003004
+Barriers: -1304 [1306=>2]
+
+Avatar_-_The_Last_Airbender_SLUS-21395_20240428221833
+Barriers: -999 [17716=>16717]
+```
+
+:::info
+The less barriers there is, the better!
+:::
+
+#### More, more!
+
+Following the adjustments by TellowKrinkle, LightningTerror has discovered that we can further cut down the barriers on OpenGL in [#13576](https://github.com/PCSX2/pcsx2/pull/13576) to only issue barriers for framebuffer optimizations if needed. To check if the state changed previous draw/pass and if not, only then do we need to issue a barrier or render pass to ensure writes.
+
+We basically have an optimization where we replace render passes with barriers in some cases to ensure writes happen, but there were cases where we issued both at the same time, so in this case if we issued a render pass state change then we don't really need barriers.
+
+Here is even more barrier count reductions that we found:
+
+```ini
+Ace_Combat_04_-_Shattered_Skies_SLUS-20152_20230214003004
+Barriers: -1302 [1306=>4]
+
+Final Fantasy XII preload frame data texture issues
+Barriers: -1144 [2291=>1147]
+
+The Getaway _PAL-M5__SCES-51159
+Barriers: -1001 [2002=>1001]
+
+Keroro Gunsou - Mero Mero Battle Royale_SLPS-25399_20230724102648
+Barriers: -1530 [3071=>1541]
+
+Need for Speed - Carbon_SLUS-21493_20220727201009
+Barriers: -2073 [3769=>1696]
+
+Need_for_Speed_-_Most_Wanted_Black_Edition_SLES-53857_20230808153118
+Barriers: -2125 [4250=>2125]
+
+Stuntman_SLUS-20250_20230326220301
+Barriers: -2026 [4053=>2027]
+
+Transformers_Special_Edition_SLES-52388_20240618185125
+Barriers: -2160 [4320=>2160]
+
+WRC_4_SCES-52389_20230329024724
+Barriers: -6603 [13473=>6870]
+```
+
+#### MORE!
+
+In [#12969](https://github.com/PCSX2/pcsx2/pull/12969), Refraction found a way to improve dirty and readback handling.
+
+Dirty targets were not combining if they overlapped in one direction, which caused there to be extra dirty uploads. Readbacks were happening on 24bit targets when the alpha channel was being requested, this avoids such situations in all renderers.
```

Comment:
I don't know what this means. Nor would a casual reader. We need to balance a blog post style with a changelog style.